internal/http/services/owncloud/ocdav/dav.go

```diff
@@ -316,12 +347,20 @@ func (h *DavHandler) Handler(s *svc) http.Handler {
 			}
 			log.Debug().Interface("statInfo", sRes.Info).Msg("Stat info from public link token path")
 
-			ctx := ContextWithTokenStatInfo(ctx, sRes.Info)
-			r = r.WithContext(ctx)
-			if sRes.Info.Type != provider.ResourceType_RESOURCE_TYPE_CONTAINER {
-				h.PublicFileHandler.Handler(s).ServeHTTP(w, r)
+			if isInternal {
+				base := path.Join(ctx.Value(net.CtxKeyBaseURI).(string), "spaces")
+				ctx := context.WithValue(ctx, net.CtxKeyBaseURI, base)
+				r = r.WithContext(ctx)
+				r.URL.Path = "/" + storagespace.FormatResourceID(*sRes.GetInfo().GetId())
+				h.SpacesHandler.Handler(s, nil).ServeHTTP(w, r)
```

Comment:
Hm, I guess this is a bit of a misunderstanding. By:

"authenticated clients accessing a pubic link (password protected or not) for a resource they already have access to are also redirected to the "real" resource. (and always need to supply the password)"

With "redirected". We were indeed referring to a HTTP Redirect response (Status 30x) to  `/dav/files/<fileid>`. It wasn't the intention to just forward the request internally. (Unless something with that has changed.